buildSrc/src/main/minimumRuntime/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -31,65 +29,47 @@
-                        stderr = new ByteArrayOutputStream();
-                        readStdout = () -> ((ByteArrayOutputStream) stdout).toString(StandardCharsets.UTF_8);
-                        readStderr = () -> ((ByteArrayOutputStream) stderr).toString(StandardCharsets.UTF_8);
+        if (getLogger().isInfoEnabled() == false) {
+            setIgnoreExitValue(true);
+            setSpoolOutput(false);
+            doLast(task -> {
+                if (getExecResult().getExitValue() != 0) {
+                    try {
+                        getLogger().error("Output for " + getExecutable() + ":");
+                        getLogger().error(outputReader.get());
+                    } catch (Exception e) {
+                        throw new GradleException("Failed to read exec output", e);
                     }
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
+                    throw new GradleException(
+                        String.format(
+                            "Process '%s %s' finished with non-zero exit value %d",
+                            getExecutable(),
+                            getArgs(),
+                            getExecResult().getExitValue()
+                        )
+                    );
                 }
-
-                setStandardOutput(stdout);
-                setErrorOutput(stderr);
-                setIgnoreExitValue(true);
-                doLast(task -> {
-                    if (getExecResult().getExitValue() != 0) {
-                        try {
-                            getLogger().error("Standard output:");
-                            getLogger().error(readStdout.get());
-                            getLogger().error("Standard error:");
-                            getLogger().error(readStderr.get());
-                        } catch (Exception e) {
-                            throw new GradleException("Failed to read exec output", e);
-                        }
-                        throw new GradleException(
-                            String.format(
-                                "Process '%s %s' finished with non-zero exit value %d",
-                                getExecutable(),
-                                getArgs(),
-                                getExecResult().getExitValue()
-                            )
-                        );
-                    }
-                });
             });
         }
     }
 
-    public void setUseFileBuffer(boolean useFileBuffer) {
-        this.useFileBuffer.set(useFileBuffer);
+    @Internal
+    public void setSpoolOutput(boolean useFileBuffer) {
+        final OutputStream out;
+        if (useFileBuffer) {
+            File spoolFile = new File(getProject().getBuildDir() + "/buffered-output/" + this.getName());
+            out = new LazyFileOutputStream(spoolFile);
+            outputReader = () -> Files.readString(spoolFile.toPath());
```

Comment:
Another potential optimization (and not a blocker) would be to define instead an `outputLogger` instead of `outputReader` which also handled the logging bit. This way the file-based once could stream from the spool file to the console instead of having to read the entire log file into memory.